### PR TITLE
86bzb002e roles microfrontend app

### PR DESCRIPTION
**Pull request by brett-onions**
## Summary by CodeRabbit

- **New Features**
	- Introduced a complete client management system with components for adding, editing, and listing clients.
	- Added dynamic role management with interfaces for handling client roles.
	- Implemented a responsive user interface using Material-UI for seamless navigation and interaction.

- **Bug Fixes**
	- Enhanced routing structure for smoother navigation between client management views.

- **Documentation**
	- Added configuration files for ESLint, Babel, Jest, TypeScript, and Webpack to ensure a standardized development environment.

- **Style**
	- Improved application styling with new CSS rules and layout adjustments for better user experience.

- **Tests**
	- Established unit tests for key components to ensure functionality and reliability.

### CODE REVIEW

File: packages/clients-app/src/pages/add-client/add-client.tsx [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+      <Grid item xs={12}>
+        <Typography variant="h3" fontSize={'32px'} fontWeight={400}>
+          Add Client
+        </Typography>
+
+        <p style={{opacity: 0.6, fontSize: '16px'}}>
+          Control client systems and their access roles. Add clients to enable
+          their request routing and group them by roles for streamlined channel
+          access management
+        </p>
```
Comment (marvinIsSacul) [inline packages/clients-app/src/pages/add-client/add-client.tsx:210]: Perhaps?

```suggestion
        <Typography variant='body1' component={"p"} style={{opacity: 0.6, fontSize: '16px'}}>
          Control client systems and their access roles. Add clients to enable
          their request routing and group them by roles for streamlined channel
          access management
        </Typography>
```

**Review by marvinIsSacul — APPROVED** [review]
LGTM with some minor suggestions.

**Review by marvinIsSacul — CHANGES_REQUESTED** [review]
Just need to fix the failing pipeline.

**Comment by drizzentic** [comment]
@coderabbitai review

File: packages/client-roles-app/.eslintrc [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "extends": [
+    "ts-react-important-stuff",
+    "plugin:prettier/recommended"
+  ],
+  "parser": "@babel/eslint-parser"
+}
```
[the remark this answers is not in the document]
Comment (brett-onions) [reply packages/client-roles-app/.eslintrc:7]: this is something to look into later on

File: packages/clients-app/src/pages/add-client/add-client.tsx [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+          enqueueSnackbar(error.response.data, {variant: 'error'})
+        } else {
+          console.log(JSON.stringify(error))
+          enqueueSnackbar('Error while creating new client', {variant: 'error'})
+        }
+      })
+      .finally(() => {
+        returnToClientList()
+      })
+  }
```
[the remark this answers is not in the document]
Comment (brett-onions) [reply packages/clients-app/src/pages/add-client/add-client.tsx:130]: Will handle in the client app pr

**Pull request merged**